cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/worker/TaskWorkerMetricsTest.java

```diff
@@ -0,0 +1,160 @@
+import io.cdap.cdap.api.service.worker.RunnableTaskRequest;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.conf.SConfiguration;
+import io.cdap.cdap.common.http.DefaultHttpRequestConfig;
+import io.cdap.cdap.metrics.collect.AggregatedMetricsCollectionService;
+import io.cdap.common.http.HttpRequest;
+import io.cdap.common.http.HttpRequests;
+import io.cdap.common.http.HttpResponse;
+import org.apache.twill.discovery.InMemoryDiscoveryService;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Tests for TaskWorker Metrics
+ */
+public class TaskWorkerMetricsTest {
+
+  private static final Gson GSON = new Gson();
+
+  private TaskWorkerService taskWorkerService;
+  private List<MetricValues> published;
+  private URI uri;
+
+  private CConfiguration createCConf() {
+    CConfiguration cConf = CConfiguration.create();
+    cConf.set(Constants.TaskWorker.ADDRESS, "localhost");
+    cConf.setInt(Constants.TaskWorker.PORT, 0);
+    cConf.setBoolean(Constants.Security.SSL.INTERNAL_ENABLED, false);
+    cConf.set(Constants.TaskWorker.PRELOAD_ARTIFACTS, "");
+    cConf.setInt(Constants.TaskWorker.CONTAINER_KILL_AFTER_REQUEST_COUNT, 1);
+    return cConf;
+  }
+
+  @Before
+  public void beforeTest() {
+    CConfiguration cConf = createCConf();
+    SConfiguration sConf = SConfiguration.create();
+    published = new ArrayList<>();
+
+    AggregatedMetricsCollectionService mockMetricsCollector = new AggregatedMetricsCollectionService(1000L) {
+      @Override
+      protected void publish(Iterator<MetricValues> metrics) {
+        Iterators.addAll(published, metrics);
+      }
+    };
+    TaskWorkerService taskWorkerService = new TaskWorkerService(cConf, sConf, new InMemoryDiscoveryService(),
+                                                                (namespaceId, retryStrategy) -> null,
+                                                                mockMetricsCollector);
+    // start the service
+    taskWorkerService.startAndWait();
+    this.taskWorkerService = taskWorkerService;
+    InetSocketAddress addr = taskWorkerService.getBindAddress();
+    this.uri = URI.create(String.format("http://%s:%s", addr.getHostName(), addr.getPort()));
+  }
+
+  @After
+  public void afterTest() {
+    if (taskWorkerService != null) {
+      taskWorkerService.stopAndWait();
+      taskWorkerService = null;
+    }
+  }
+
+  @Test
+  public void testSimpleRequest() throws IOException {
+    String taskClassName = TaskWorkerServiceTest.TestRunnableClass.class.getName();
+    RunnableTaskRequest req = RunnableTaskRequest.getBuilder(
+      taskClassName).withParam("100").build();
```

Comment:
Don't break the line at the beginning of the argument. Break it like this:

```
RunnableTaskRequest req = RunnableTaskRequest.getBuilder(taskClassName)
  .withParam("100")
  .build();
```

Reply:
Done.